src/ewasm/utils.rs

```diff
@@ -0,0 +1,37 @@
+use super::inkwell::types::AnyTypeEnum;
+use crate::ast::Identifier;
+use crate::ewasm::codegen::Codegen;
+use crate::ewasm::function_context::FunctionContext;
+
+pub fn get_num_pointer_layers(val_type: AnyTypeEnum) -> u8 {
+    let mut num_pointers = 0;
+    let mut val_type = val_type;
+    while val_type.is_pointer_type() {
+        num_pointers += 1;
+        val_type = val_type.into_pointer_type().get_element_type();
+    }
+    num_pointers
+}
+
+pub fn generate_caller_variable<'ctx>(
+    codegen: &mut Codegen<'_, 'ctx>,
+    function_context: &mut FunctionContext<'ctx>,
+    caller_binding: Option<Identifier>,
+) {
+    let caller_address = codegen
+        .builder
+        .build_call(
+            codegen.module.get_function("_getCaller").unwrap(),
+            &[],
+            "tmp_call",
```

Comment:
I assume these are magic strings?

Reply:
_getCaller is, the other one is just a name as explained elsewhere. _getCaller has now been made a constant